src/wallet/wallet.h

```diff
@@ -555,6 +555,7 @@ class CWalletTx
     // wrong copy.
     CWalletTx(CWalletTx const &) = delete;
     void operator=(CWalletTx const &x) = delete;
+    bool isFinal() const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
In commit "[wallet] Replace Chain::checkFinal by CWallet::CheckFinalWalletTx" (03870881969b12d8023716897ff674ec6ec3568e)

Travis error here:

```
In file included from interfaces/wallet.cpp:24:
./wallet/wallet.h:558:51: error: use of undeclared identifier 'cs_wallet'; did you mean 'pwallet'?
    bool isFinal() const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
                                                  ^~~~~~~~~
                                                  pwallet
./threadsafety.h:31:79: note: expanded from macro 'EXCLUSIVE_LOCKS_REQUIRED'
#define EXCLUSIVE_LOCKS_REQUIRED(...) __attribute__((exclusive_locks_required(__VA_ARGS__)))
                                                                              ^
./wallet/wallet.h:272:20: note: 'pwallet' declared here
    const CWallet* pwallet;
```

https://travis-ci.org/github/bitcoin/bitcoin/jobs/688716547#L3380

Probably needs to say `pwallet->cs_wallet` not just `cs_wallet`

I wonder @ariard if you're able to test lock annotations locally since these errors seem to come up a lot. It shouldn't be hard to enable them. I just configure with `./configure CXX=clang++ --enable-debug --enable-werror` for these

Reply:
Seeing the same messages while building locally, the gcc 8.3 build is fine but the clang 6.0 debug/werror build errors.

Reply:
Building with `pwallet->cs_wallet` halts with
```
./wallet/wallet.h:558:58: error: member access into incomplete type 'const CWallet'
    bool isFinal() const EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet);
                                                         ^
./wallet/fees.h:13:7: note: forward declaration of 'CWallet'
class CWallet;
      ^
```

Reply:
@jonatack @ryanofsky You right I didn't the warning at first, was compiling with gcc locally. Okay in fact the issue hasn't been solved wrt to `NO_THREAD_SAFETY_ANALYSIS`. 

Isn't the way to solve it cleanly would be to get `CWalletTx` its own lock instead of requiring `CWallet` one ? I don't see a better way to solve dependency declaration..

Reply:
Updated dropping `isFinal` accessor on the advice of @ryanofsky